pyspecdata/general_functions.py

```diff
@@ -138,21 +138,23 @@ def init_logging(level=logging.INFO, filename='pyspecdata.log'):
             level=logging.DEBUG
         else:
             raise ValueError("if you give me level as a string, give me 'info' or 'debug'")
+    FORMAT = "--> %(filename)s(%(lineno)s):%(name)s %(funcName)20s %(asctime)20s\n%(levelname)s: %(message)s"
+    formatter = logging.Formatter(FORMAT)
     log_filename = os.path.join(os.path.expanduser('~'),filename)
     if os.path.exists(log_filename):
         # manually remove, and then use append -- otherwise, it won't write to
         # file immediately
         os.remove(log_filename)
     logger = logging.getLogger()
-    #fileHandler = logging.FileHandler(filename)
-    stdout_handler = logging.StreamHandler(sys.stdout)        
-    stdout_handler.setLevel(level)
-    #logger.setLevel(level)
-    formatter = logging.Formatter("--> %(filename)s(%(lineno)s):%(name)s %(funcName)20s %(asctime)20s\n%(levelname)s: %(message)s")
-    #fileHandler.setFormatter(logging.Formatter(FORMAT))
+    logger.setLevel(level) # even if I set the handler level, it won't
+    #                        print w/out this
+    file_handler = logging.FileHandler(log_filename, mode='a')
```

Comment:
The problem mentioned above was that you want a FileHandler, which takes a file name, rather than a stream object (sys.stdout is a stream object)

Reply:
Okay so the logger = logging.getLogger() we NEED because we should NOT represent  loggers directly but through this module-level function (according to https://docs.python.org/3/library/logging.html).
the setLevel will set the threshold to level, so that if the logging message is less important than the level it will just be ignored. (same resource as above)
FileHandler will write the log in a file. But I believe to display it we need FileStreamer?

Reply:
so I see the StreamHandler is addressed after this and this totally makes sense now

Reply:
`getLogger()` gets the logger object that you can use to set the formatting, attach handlers, etc.

sounds like you are saying you get the difference between `StreamHandler` and `FileHandler` --> obviously, this is a one-off that you probably won't have to deal with again, so it's not about the specifics going on here --> the skill to really take away from the experience is how you look at the documentation/stackexchange/etc to figure out how to use those two functions on the fly.

Reply:
yes much better